### PR TITLE
Use CN from CSRs

### LINKED ISSUE #1076: Update --csr to check common name
Even if you're requesting a cert for a single name, the client currently only reads SANs to determine which domains are being requested. The client should also check the CN for domain names to include in the request.

### PR DESCRIPTION
**Pull request by bmw**
Fixes #1076 and brings in #2684.

### CODE REVIEW

File: certbot/crypto_util.py [hunk 1]
```diff
@@ -179,6 +180,30 @@ def csr_matches_pubkey(csr, privkey):
...
+    :param str data: contents of the CSR file
+
+    :returns: (`OpenSSL.crypto.FILETYPE_PEM` or `OpenSSL.crypto.FILETYPE_ASN1`,
+               le_util.CSR object representing the CSR,
+               list of domains requested in the CSR)
+    :rtype: tuple
+
+    """
+    for form, typ in (("der", OpenSSL.crypto.FILETYPE_ASN1,),
+                      ("pem", OpenSSL.crypto.FILETYPE_PEM,),):
```
Comment (SwartzCr) [inline certbot/crypto_util.py]: what's with this trailing comma?

File: certbot/crypto_util.py [hunk 2]
```diff
@@ -267,6 +297,25 @@ def get_sans_from_csr(csr, typ=OpenSSL.crypto.FILETYPE_PEM):
...
+    :param typ: `OpenSSL.crypto.FILETYPE_PEM` or `OpenSSL.crypto.FILETYPE_ASN1`
+
+    :returns: A list of domain names.
+    :rtype: list
+
+    """
+    loaded_csr = _load_cert_or_req(
+        csr, OpenSSL.crypto.load_certificate_request, typ)
+    # Use a set to avoid duplication with CN and Subject Alt Names
+    domains = set(d for d in (loaded_csr.get_subject().CN,) if d is not None)
```
Comment (SwartzCr) [inline certbot/crypto_util.py]: and this trailing comma

File: certbot/main.py [hunk 3]
```diff
@@ -681,9 +681,6 @@ def main(cli_args=sys.argv[1:]):
         displayer = display_util.NoninteractiveDisplay(sys.stdout)
     elif config.text_mode:
         displayer = display_util.FileDisplay(sys.stdout)
-    elif config.verb == "renew":
```
Comment (SwartzCr) [inline certbot/main.py]: where is this addressed now?

File: certbot/crypto_util.py (hunk 2 above)
Comment (bmw) [inline certbot/crypto_util.py]: This one is necessary ([source](https://stackoverflow.com/questions/12876177/why-do-tuples-with-only-one-element-get-converted-to-strings)).

File: certbot/crypto_util.py (hunk 1 above)
Comment (bmw) [inline certbot/crypto_util.py]: This one is stylistic, there for consistency, and because it makes me happy.

File: certbot/main.py (hunk 3 above)
Comment (bmw) [inline certbot/main.py]: [here](https://github.com/certbot/certbot/pull/3046/files/5cb0e0f264ef94187b774561329d2ab432e2634e#diff-98e51bbf59cb44db5dc06a7c5a002983R337)

**Comment by SwartzCr** [comment]
lgtm

**Pull request merged**